### PR TITLE
STL-393 - PoET Spec

### PR DESCRIPTION
**Pull request by aludvik**
This is a roughly 1-to-1 transposition of the PoET spec doc to sphinx.

I have also written a PoET primer which I have not converted to sphinx at: https://docs.google.com/a/intel.com/document/d/1se5sye23o8muot-TrYSK4Yl_Ue92Uh4yf5dHLSBt4_Y/edit?usp=sharing

I think this doc would be good to add as well (in another PR), either as another standalone doc or integrated into the spec.

### CODE REVIEW

**Comment by aludvik** [comment]
HTML can be found here: https://build.sawtooth.me/job/Sawtooth-Hyperledger/job/sawtooth-core/job/PR-697/1/artifact/docs/build/html/architecture/poet.html

File: docs/source/architecture/poet.rst [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+
+**Parameters**
+
+.. code:: console
+
+    byte[672] sealedSignUpData # (PPK, PSK, MCID) tuple encrypted with SealKey
+
+**Description**
+
+1. Use ``sgx_unseal_data()`` function to decrypt sealedSignUpData into (PPK,
```
Comment (jsmitchell) [inline docs/source/architecture/poet.rst]: It would be good to review all this type of text and add articles "Use _the_ function", etc.

**Comment by aludvik** [comment]
Addresses @jsmitchell's requests.

File: docs/source/architecture/poet.rst [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+PoET 1.0 Specification
+**********************
+
+Introduction
+============
+
+The Proof-of-Elapsed-Time or PoET Consensus method offers a solution to the
+Byzantine Generals Problem that utilizes a “trusted execution environment” to
+improve on the efficiency of present solutions such as Proof-of-Work. The
+following presentation assumes the use of Intel SGX as the trusted execution
```
Comment (dcmiddle) [inline docs/source/architecture/poet.rst]: Before the last sentence please insert:
The initial release of PoET was fully abstracted so as to be portable to any trusted execution environment.

File: docs/source/architecture/poet.rst [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+
+  The term can also be used to refer to a specific enclave that has been
+  initialized with a specific code and data.
+
+Basename
+  A service provider base name. In our context the service provider
+  entity is the distributed ledger network. Each distinct network should have
+  its own Basename and Service Provider ID (see EPID and IAS specifications).
+
+EPID Pseudonym
```
Comment (dcmiddle) [inline docs/source/architecture/poet.rst]: Add 
```
EPID
  An anonymous credential system. See E. Brickell and Jiangtao Li: “Enhanced Privacy ID from Bilinear Pairing for Hardware Authentication and Attestation”. IEEE International Conference on Social Computing / IEEE International Converence on Privacy, Security, Risk and Trust. 2010.```

**Comment by randombishop** [comment]
Great contribution, was looking exactly for that !

**Comment by aludvik** [comment]
Addressed Dan's comments.

**Pull request merged**